violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java

```diff
@@ -0,0 +1,256 @@
+                path.moveTo(ARM_WIDTH /2,0);
+                path.lineTo(ARM_WIDTH /2, TOP_NOTCH_HEIGHT);
+
+                //The upper arm
+                path.lineTo(0, TOP_NOTCH_HEIGHT);
+                path.lineTo(0, TOP_ARM_HEIGHT);
+                path.lineTo(ARM_WIDTH, TOP_ARM_HEIGHT);
+                path.lineTo(ARM_WIDTH, TOP_NOTCH_HEIGHT);
+                path.lineTo(ARM_WIDTH /2, TOP_NOTCH_HEIGHT);
+
+                //The distance between the arms
+                path.moveTo(ARM_WIDTH /2, TOP_ARM_HEIGHT);
+                path.lineTo(ARM_WIDTH /2, BOTTOM_ARM_HEIGHT);
+
+                //The bottom arm
+                path.lineTo(0, BOTTOM_ARM_HEIGHT);
+                path.lineTo(0, BOTTOM_NOTCH_HEIGHT);
+                path.lineTo(ARM_WIDTH, BOTTOM_NOTCH_HEIGHT);
+                path.lineTo(ARM_WIDTH, BOTTOM_ARM_HEIGHT);
+                path.lineTo(ARM_WIDTH /2, BOTTOM_ARM_HEIGHT);
+
+                //The bottom notch
+                path.moveTo(ARM_WIDTH /2, BOTTOM_NOTCH_HEIGHT);
+                path.lineTo(ARM_WIDTH /2, contentHeight);
+
+                //The connection of the node
+                path.lineTo(contentWidth, contentHeight);
+                path.lineTo(contentWidth, 0);
+                path.lineTo(ARM_WIDTH /2, 0);
+
+                return path;
+            }
+        });
+        setBorder(new ContentBorder(contentInsideShape, getBorderColor()));
+        setBackground(new ContentBackground(getBorder(), getBackgroundColor()));
+        setContent(getBackground());
+        setTextColor(super.getTextColor());
+    }
+
+    /**
+     * Sets the node text color.
+     */
+    @Override
+    public void setTextColor(Color textColor)
+    {
+        text.setTextColor(textColor);
+        super.setTextColor(textColor);
+    }
+
+    @Override
+    public String getToolTip()
+    {
+        return tooltip;
+    }
+
+    @Override
+    public LineText getName()
+    {
+        return text;
+    }
+
+    @Override
+    public LineText getAttributes()
+    {
+        return null;
+    }
+
+    @Override
+    public LineText getMethods()
+    {
+        return null;
+    }
+
+    /**
+     * Gets connection point.
+     */
+    @Override
+    public Point2D getConnectionPoint(IEdge edge)
+    {
+        Point2D connectionPoint = super.getConnectionPoint(edge);
+
+        if (Direction.EAST.equals(edge.getDirection(this).getNearestCardinalDirection()))
+        {
+            double offset = MIN_HEIGHT / 2 - Math.abs(connectionPoint.getY() - getLocation().getY() - MIN_HEIGHT / 2);
+
+            return new Point2D.Double(
+                    connectionPoint.getX() + offset,
+                    connectionPoint.getY()
+            );
+        }
+
+        return connectionPoint;
+    }
+
+    /**
+     * Adds connection edge.
+     */
+    @Override
+    public boolean addConnection(IEdge edge)
+    {
+        if (edge.getEndNode() != null && this != edge.getEndNode())
+        {
+            return true;
```

Comment:
I don't like double returns in one method.
It should be better, if you change it to one return.
boolean result = false;
if(...) {
result = true;
}
return result;